### PR TITLE
Add a way to specify the script name to run cron:run command from non…

### PR DESCRIPTION
**Pull request by Pixelshaped**
…-CLI contexts

This PR is in response to this issue https://github.com/Cron/Symfony-Bundle/issues/89

It implements a way to specify the script name to run the command from non-CLI context (i.e. apache2handler, etc.)

That way, it's possible to call the cron:run command from a controller, on hostings where it's needed.
```
<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Console\Application;
use Symfony\Component\Console\Input\ArrayInput;
use Symfony\Component\Console\Output\BufferedOutput;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\HttpKernel\KernelInterface;
use Symfony\Component\Routing\Annotation\Route;

class CronController extends BaseAppController
{
    /**
     * @Route("/cron/run", name="cron_run_endpoint")
     */
    public function endpoint(KernelInterface $kernel): Response
    {
        $application = new Application($kernel);
        $application->setAutoExit(false);

        $input = new ArrayInput([
            'command' => 'cron:run',
            '--script_name' => 'bin/console',
        ]);

        $output = new BufferedOutput();

        $application->run($input, $output);

        $content = $output->fetch();

        return new Response($content);
    }
}

```

Please let me know if you need anything more to accept and merge the PR.

### CODE REVIEW

File: Command/CronRunCommand.php [hunk 1]
```diff
@@ -35,7 +35,8 @@ protected function configure()
             ->setDescription('Runs any currently schedule cron jobs')
             ->addArgument('job', InputArgument::OPTIONAL, 'Run only this job (if enabled)')
             ->addOption('force', null, InputOption::VALUE_NONE, 'Force schedule the current job.')
-            ->addOption('schedule_now', null, InputOption::VALUE_NONE, 'Temporary set the job schedule to now.');
+            ->addOption('schedule_now', null, InputOption::VALUE_NONE, 'Temporary set the job schedule to now.')
```
Comment (NoUseFreak) [inline Command/CronRunCommand.php:38]: Please change the option's word separator with `-` instead of `_`.

File: Command/CronRunCommand.php [hunk 2]
```diff
@@ -35,7 +35,8 @@ protected function configure()
...
+            ->addOption('script_name', null, InputOption::VALUE_OPTIONAL, 'Specify this to avoid guessing the script name to run the command in non-CLI context.');
```
Comment (NoUseFreak) [inline Command/CronRunCommand.php:39]: Please change the option's word separator with `-` instead of `_`.

**Review by NoUseFreak — CHANGES_REQUESTED** [review]
Hi @Pixelshaped 

Thanks for the addition. Could you fix the minor remark I left in the review and resolve the merge conflict, than we can merge this.

File: Command/CronRunCommand.php (hunk 1 above)
Comment (Pixelshaped) [reply Command/CronRunCommand.php:38]: OK I did it but shouldn't `schedule_now` follow the same pattern?

**Pull request merged**